src/sentry/snuba/metrics/extraction.py

```diff
@@ -905,12 +936,28 @@ def tags_conditions(self, project: Project) -> List[TagSpec]:
 
         return tags_specs_generator(project, self._arguments)
 
+    def _tag_for_column(self, column: str) -> TagSpec:
+        """Returns a TagSpec for a dynamic column"""
+        field = _map_field_name(column)
+
+        return {
+            "key": field,
```

Comment:
Another small thing, if we keep the column name as the key, we might leverage the column resolution of the query builders but I am not 100% sure it could be plugged in easily. 

How you did it it's perfectly fine I am just thinking out loud.

Reply:
So I was thinking keeping key=field so we know exactly was attempted for extraction into the field _but_ I think you're right since they we don't update the QueryBuilders to just use the intended column. Also we'll be making a new set of indexer values when a lot of them already exist for MEP data.

Reply:
That's a good idea. At the end of the day we will have our own custom logic for the query resolution, so it's not a big deal.

Actually, it's even better since in a nicer future we would like to lift out on-demand metrics out of the query builder but this is a utopian future.